pkg/config/validation.go

```diff
@@ -209,11 +209,25 @@ func validatePodIntegrationOptions(c *configapi.Configuration) field.ErrorList {
 		return allErrs
 	}
 
-	if c.Integrations.PodOptions == nil {
-		return field.ErrorList{field.Required(podOptionsPath, "cannot be empty when pod integration is enabled")}
+	var namespaceSelector *metav1.LabelSelector = nil
+	var namespaceSelectorPath *field.Path
+	if features.Enabled(features.ManagedJobsNamespaceSelector) {
+		namespaceSelectorPath = managedJobsNamespaceSelectorPath
+		if c.Integrations.PodOptions != nil && c.Integrations.PodOptions.NamespaceSelector != nil {
+			namespaceSelector = c.Integrations.PodOptions.NamespaceSelector
+			namespaceSelectorPath = podOptionsNamespaceSelectorPath
+		} else if c.ManagedJobsNamespaceSelector != nil {
+			namespaceSelector = c.ManagedJobsNamespaceSelector
+		}
+	} else {
```

Comment:
```suggestion
```

Should we validate podOptions even if ManagedJobsNamespaceSelector?
Because we use it when it is enabled: https://github.com/kubernetes-sigs/kueue/pull/4256/files#diff-64a1e0bd9f87bdadd12179b8e48ace63c6aa3ace3f7694375a8a0dd0e566e25dR336-R338

Reply:
Adjusted so that we always validate podOptions if it is non-nil.